examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,338 @@
+        // If you don't intend on exhaustively specifying an operator() for each
+        // event type, you can provide a templated operator() that will be selected
+        // by overload resolution when no other event type matches.
+        template <typename T>
+        std::optional<std::string> operator()(const T&)
+        {
+            // All unhandled events will end up here
+            // application.m_log.emplace_back("Other Event");
+            return std::nullopt;
+        }
+
+        Application& application;
+    };
+
+    ////////////////////////////////////////////////////////////
+    void run()
+    {
+        // This example is for demonstration purposes only
+        // All the following forms of event handling have equivalent behavior
+        // In your own code you should decide which form of event handling
+        // suits your needs best and use a single form of event handling
+        while (m_window.isOpen())
+        {
+            if (m_handlerType == HandlerType::Classic)
+            {
+                // The "classical form" of event handling
+                // Poll/Wait for events in a loop and handle them
+                // individually based on their concrete type
+                while (const std::optional event = m_window.pollEvent())
+                {
+                    if (event->is<sf::Event::Closed>())
+                    {
+                        m_window.close();
+                        break;
+                    }
+
+                    if (const auto* keyPress = event->getIf<sf::Event::KeyPressed>())
+                    {
+                        m_log.emplace_back("Key Pressed: " + sf::Keyboard::getDescription(keyPress->scancode));
+
+                        // When the enter key is pressed, switch to the next handler type
+                        if (keyPress->code == sf::Keyboard::Key::Enter)
+                        {
+                            m_handlerType = HandlerType::Visitor;
+                            m_handlerText.setString("Current Handler: Visitor");
+                        }
+                    }
+                    else if (const auto* mouseMoved = event->getIf<sf::Event::MouseMoved>())
+                    {
+                        m_log.emplace_back("Mouse Moved: " + vec2ToString(mouseMoved->position));
+                    }
+                    else if (event->is<sf::Event::MouseButtonPressed>())
+                    {
+                        m_log.emplace_back("Mouse Pressed");
+                    }
+                    else if (const auto* touchBegan = event->getIf<sf::Event::TouchBegan>())
+                    {
+                        m_log.emplace_back("Touch Began: " + vec2ToString(touchBegan->position));
+                    }
+                    else
+                    {
+                        // All unhandled events will end up here
+                        // m_log.emplace_back("Other Event");
```

Comment:
Why comment it out?

Reply:
To reduce spam by a tiny bit, and to encourage experimentation.